app/uk/gov/hmrc/tradergoodsprofilesrouter/connectors/EisHttpErrorHandler.scala

```diff
@@ -35,8 +35,26 @@ trait EisHttpErrorHandler extends Logging {
     httpResponse.status match {
 
       case BAD_REQUEST =>
-        EisHttpErrorResponse(BAD_REQUEST, determine400Error(correlationId, httpResponse.body))
-      case FORBIDDEN   =>
+        val checkError            = determine400Error(correlationId, httpResponse.body)
```

Comment:
HI, good working solution. This may be more cleare if refactored as following:

```suggestion
       val checkError: ErrorResponse = determine400Error(correlationId, httpResponse.body)

        checkError
          .errors
          .flatMap(_.filter(_.errorNumber == 7).headOption) match {
          case Some(_) =>
            EisHttpErrorResponse(FORBIDDEN, ErrorResponse(correlationId,ForbiddenCode,ForbiddenMessage,checkError.errors))
          case _ => EisHttpErrorResponse(BAD_REQUEST, checkError)
        }
```